### PR TITLE
tier1, Add kubernetes nmstate tier1 automation test

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
This PR adds a script that runs nmstate e2e test on CNAO cluster

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RamLavi** [comment]
/hold

need to replace unwanteduse of `KUBECTL=kubectl`

**Comment by RamLavi** [comment]
/unhold
fixed use of kubectl

**Comment by RamLavi** [comment]
/hold

This PR depends on #635  to be merged

File: automation/check-patch.e2e-nmstate-functests.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    trap teardown EXIT
+
+    echo "Configure test parameters"
+    export TIMEOUT=${TIMEOUT:-1h}
+    export NAMESPACE=${NAMESPACE:-cluster-network-addons}
+    export KUBECTL=${KUBECTL:-${TMP_PROJECT_PATH}/cluster/kubectl.sh}
+    export SSH=${SSH:-"${TMP_PROJECT_PATH}/cluster/ssh.sh"}
+    export CLUSTER_PATH=${CLUSTER_PATH:-${TMP_PROJECT_PATH}/_kubevirtci/}
+
+    cat << EOF > ${SSH}
```
Comment (RamLavi) [inline automation/check-patch.e2e-nmstate-functests.sh:39]: @qinqon this part rewriting the current content of ssh.sh makes the script get stuck:
```
kubernetes-nmstate (master) $ $SSH
+ node_name=
++ CLUSTER_PATH=/tmp/cnao/cluster-network-addons-operator/_kubevirtci/
++ /tmp/cnao//cluster-network-addons-operator/cluster/kubectl.sh get node --no-headers -o wide
++ awk '{print $6}'
+ node_ip='192.168.66.101
192.168.66.102
192.168.66.103'
+ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null core@192.168.66.101 192.168.66.102 192.168.66.103 --
```

Comment (RamLavi) [reply automation/check-patch.e2e-nmstate-functests.sh]: removed it, as discussed offline

File: automation/check-patch.e2e-nmstate-functests.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    # this script also exports KUBECONFIG, and fetch $COMPONENT repository
+    export KUBEVIRT_NUM_NODES=3 # 1 master, 2 workers
+    export KUBEVIRT_NUM_SECONDARY_NICS=2
+    COMPONENT="nmstate" source automation/components-functests.setup.sh
+
+    trap teardown EXIT
+
+    echo "Configure test parameters"
+    export TIMEOUT=${TIMEOUT:-1h}
+    export NAMESPACE=${NAMESPACE:-cluster-network-addons}
```
Comment (phoracek) [inline automation/check-patch.e2e-nmstate-functests.sh:34]: We don't need any of these variables to be configurable. The role of `automation/` scripts is to run on any machine with zero configuration. It should be self sufficient getting its own cluster and its own Go.

**Review by phoracek — CHANGES_REQUESTED** [review]

File: automation/check-patch.e2e-nmstate-functests.sh (hunk 2 above)
Comment (RamLavi) [reply automation/check-patch.e2e-nmstate-functests.sh:34]: so you're saying I shouldn't I shouldn't export them?or not even pass these variables to the `make test-e2e-handler` command?

Comment (phoracek) [reply automation/check-patch.e2e-nmstate-functests.sh:34]: Keep exporting them. There is just no need to make them configurable:

Turn `${TIMEOUT:-1h}` to `1h`

Comment (RamLavi) [reply automation/check-patch.e2e-nmstate-functests.sh]: ack.
[Done](https://github.com/kubevirt/cluster-network-addons-operator/compare/284194c866d30059385fbb563e03094a087cbab8..2c31d617a8d910a39f73ecdfc1c193e4a53bc32e)

**Review by phoracek — APPROVED** [review]
/lgtm
/approve

**Pull request merged**